File: testing/c/test_c.py
from wasmite import WasiModule, WasmiteCase, main
from wasmite import FuncType, Function
from wasmite import I32, I64


class Test(WasmiteCase):
    module = WasiModule("test_c.wasm")
    exports = module.exports
    
    def test_main(self):
        # test and check main function
        self.module.run_main()
    
    def test_even(self):
        # test the "even" function in test_cpp.wasm
        result = self.exports.even(1)
        self.assertEqual(result, 0) 
        
    def test_squared(self):
        # test the "squared" function in test_cpp.wasm
        result = self.exports.squared(1) # 2**3
        self.assertEqual(result, 1)
        
# Hi don't forget to add me         
if __name__ == "__main__":
    main()

File: testing/c/test_c.c
#include <stdio.h>

// emcc test_c.c -o test_c.wasm -s EXPORTED_FUNCTIONS='["_even", "_squared"]'
// functions even and squared are exported ------------^^^^^^^^^^^^^^^^^^^^^

int even(int value){

    if (value % 2 == 0) {
        return 1; // true: 1
    } else {
        return 0; // false: 0
    }
}

int squared(int value){
    return value * value;
}

int main() {
    int number = even(3);
    printf("even(3): %d\n", number);

    int number2 = squared(3);
    printf("squared(3) %d\n", number2);

    return 0;
}